File: src/cpp/reOrderArray.cpp
class Solution {
public:
    void reOrderArray(vector<int> &array) {
        vector<int> odd;
        vector<int> even;

        for (auto &i : array) {
            if (i & 1) {
                odd.push_back(i);
            } else {
                even.push_back(i);
            }
        }

        for (int i = 0; i < odd.size(); ++i) {
            array[i] = odd[i];
        }

        for (int i = odd.size(); i < odd.size() + even.size(); ++i) {
            array[i] = even[i - odd.size()];
        }
    }
};

File: src/java/ReOrderArray.java
public class Solution {
    public void reOrderArray(int [] array) {
        // like insertion sort
        for (int i = 0; i < array.length; i++) {
            if (array[i] % 2 == 1) {
                for (int j = i; j > 0; j--) {
                    if (array[j - 1] % 2 == 0) {
                        int temp = array[j];
                        array[j] = array[j - 1];
                        array[j - 1] = temp;
                    }
                }
            }
        }
    }
}
